src/store/productState.ts

```diff
@@ -0,0 +1,12 @@
+import { create } from 'zustand'
+import { Product } from '../model/product'
+
+interface ProductState {
+  selectedItem: Product | null
```

Comment:
Is this not derived state?

Reply:
yes. I'll change it to store the Id and get the product from that